src/Microsoft.AspNet.Server.Kestrel/Infrastructure/KestrelThread.cs

```diff
@@ -50,16 +42,10 @@ public Task StartAsync()
 
         public void Stop(TimeSpan timeout)
         {
-            Post(OnStop, null);
+            Post(OnStop);
             if (!_thread.Join((int)timeout.TotalMilliseconds))
             {
-                Post(OnStopImmediate, null);
-                if (!_thread.Join((int)timeout.TotalMilliseconds))
-                {
-#if ASPNET50
-                    _thread.Abort();
-#endif
-                }
+                throw new TimeoutException("Loop did not close");
```

Comment:
There are two reasons:

First, aborting a thread doesn't work while it's in a native call. This means while in the loop and waiting to invoke callbacks, `Abort` only schedules the `ThreadAbortException` to be called once the loop enters managed code (in either direction).

On the other hand, the documentation for Libuv is clear in that the loop will only run while there are active handles. `OnStop` makes sure that those are forcibly closed, so the loop will exit naturally.